rxandroid/src/main/java/rx/android/schedulers/AndroidSchedulers.java

```diff
@@ -18,27 +18,16 @@
 import android.os.Looper;
 import rx.android.plugins.RxAndroidPlugins;
 
-/**
- * Schedulers that have Android-specific functionality
- */
+/** Android-specific Schedulers. */
 public final class AndroidSchedulers {
     private AndroidSchedulers() {
         throw new AssertionError("No instances");
     }
 
     private static final Scheduler MAIN_THREAD_SCHEDULER =
-            new HandlerThreadScheduler(new Handler(Looper.getMainLooper()));
-
-    /**
-     * {@link Scheduler} which uses the provided {@link Handler} to execute actions.
-     */
-    public static Scheduler handlerThread(final Handler handler) {
-        return new HandlerThreadScheduler(handler);
-    }
+            new HandlerScheduler(new Handler(Looper.getMainLooper()));
 
-    /**
-     * {@link Scheduler} which will execute actions on the Android UI thread.
-     */
+    /** A {@link Scheduler} which executes actions on the Android UI thread. */
     public static Scheduler mainThread() {
         Scheduler scheduler =
                 RxAndroidPlugins.getInstance().getSchedulersHook().getMainThreadScheduler();
```

Comment:
Makes me sad :( 

For just resolving `mainThread` scheduler we have to do calls to 3 virtual methods and then execute conditional check. This hook probably will be used by 1% of library users and it will give performance reduction (small, but, we are on mobile devices) for 99% other users. 

May be we can offer something like "RxAndroidPlugins.enableHooks()" which will change some boolean variable and we will check it (via static getter or whatever) before trying to resolve things via hooks? 

I understand that this is not clean solution, tho it's more likely "just saying", but I just wanted to say that RxJava already adds some overhead in compare to regular async code, so it would be nice to minimize such overhead in RxAndroid.